cloudpickle/cloudpickle.py

```diff
@@ -69,6 +69,39 @@
     from io import BytesIO as StringIO
     PY3 = True
 
+
+try:
+    from ctypes import pythonapi, py_object, c_int
+except ImportError:
+    supports_recursive_closure = False
+
+    def compress_closure(closure):
+        return closure
+
+    def decompress_closure(compressed_closure):
+        return compressed_closure
+
+    def fill_cells(cells, values):
+        pass
+else:
+    supports_recursive_closure = True
+
+    def compress_closure(closure):
+        return len(closure)
+
+    def decompress_closure(closure):
+        return tuple(_make_cell(None) for _ in range(closure))
+
+    _cell_set = pythonapi.PyCell_Set
+    _cell_set.argtypes = (py_object, py_object)
+    _cell_set.restype = c_int
```

Comment:
I changed my code a few minutes before this to do this slightly differently since it's best not to modify members in another module. See the updated version [here](https://github.com/ray-project/cloudpickle/blob/982da06bb13c330bac3e0a55fabd319778bd0c4c/cloudpickle/cloudpickle.py#L82).
(Sorry for the comment repost, GitHub deleted two comments instead of one.)